connection.ts

```diff
@@ -0,0 +1,174 @@
+import { BufReader, BufWriter } from "./vendor/https/deno.land/std/io/bufio.ts";
+import { CommandExecutor, RedisRawReply, muxExecutor, sendCommand } from "./io.ts";
+
+type Reader = Deno.Reader;
+type Writer = Deno.Writer;
+type Closer = Deno.Closer;
+
+export type RedisConnectOptions = {
+  hostname?: string;
+  port?: number | string;
+  tls?: boolean;
+  db?: number;
+  password?: string;
+  name?: string;
+  maxRetryCount?: number;
+};
+
+export class RedisConnection {
+
+  name: string | null = null
+  closer: Closer | null = null
+  executor: CommandExecutor | null = null;
+  reader: Reader | null = null;
```

Comment:
Can you change the type of `reader` to `BufReader` and `writer` to `BufWriter`?
```suggestion
  reader: BufReader;
  writer: BufWriter;
```